src/global.h

```diff
@@ -135,6 +136,8 @@ typedef struct {
 
 	StageInfo *stage;
 
+	Trainer tnr;
```

Comment:
The global object is bad enough; please rename this field to `trainer` for clarity.